docs/source/task_tutorial.rst

```diff
@@ -0,0 +1,216 @@
+
+More information on this format can be found in the documentation on ``data_loader`` in `DialogueData <http://parl.ai/static/docs/dialog.html#parlai.core.dialog_teacher.DialogData>`__ (``setup_data`` is provided as a data_loader to ``DialogueData``).
+
+The sample ``setup_data`` method for our task is presented below. 
+
+.. code-block:: python
+
+    def setup_data(self, path):
+        print('loading: ' + path)
+
+        # open data file with labels 
+        # (path will be provided to setup_data from opt['datafile'] defined above)
+        with open(path) as labels_file:
+            self.labels = json.load(labels_file)
+
+        # define standard question, since it doesn't change for this task
+        self.question = 'Which number is in the image?'
+        # every episode consists of only one query in this task
+        episode_done = True
+
+        # define iterator over all queries
+        for i in range(len(self.labels)):
+            # set up path to curent image
+            img_path = os.path.join(self.image_path, '%05d.bmp' % i)
+            # get current label, both as a digit and as a text
+            label = [self.labels[i], self.num_strs[int(self.labels[i])]]
+            # yield tuple with information and episode_done? flag
+            yield (self.question, label, None, None, img_path), episode_done
+
+As we can see from the code above, for this specific task the question is always the same, and thus it is fixed. For different tasks, this might change at each iteration. Similarly, for this task, each episode consist of only one query, thus ``episode_done?`` is always true (*i.e.*, each query is the end of its episode). This could also vary depending on the task.
+
+Looking at the tuple provided by the iterator at each yield, we can see that we defined a query, a label and an image path. When working with ``DialogTeacher`` in visual tasks, it is important to provide the path to the image in the ``setup_data`` tuple. This also allows for integrated processing (*e.g.*, conversion to ascii image by setting command line flag ``-im ascii``).
+
+Finally, one might notice that no reward or label candidates were provided in the tuple (both are set to ``None``). The reward is not specified because it is not usefult for this task. The label candidates, however, were not specified because they are always the same in this task (the digits from '0' to '9'). For cases like this, with fixed label candidates, one can simply define a method ``label_candidates()`` that returns the unchanging candidates, as demonstrated below. For cases where the label candidates vary for each query, the field in the tuple can be used.
```

Comment:
nit: usefult => useful

Maybe also "The label candidates, however, were not specified per-example for this task because we instead use a single set of universal candidates for every example in this task (the digits from '0' to '9')."